interface.js

```diff
@@ -417,6 +471,13 @@ module.exports = class Interface {
     const memory = new Uint8Array(this.module.exports.memory, offset, length)
     memory.set(value)
   }
+
+  takeGas (amount) {
+    if (this.environment.gasLimit < amount) {
+      throw new Error('Ran out of gas')
+    }
+    this.environment.gasLimit -= amount
+  }
```

Comment:
I think `useGas` should have extra logic for safety checks because it can be invoked by a contract. `takeGas` on the other hand can only be invoked from *trusted* callers.

Essentially this is removing the safety check overhead for the majority of the cases.